### PR TITLE
add docs for attaching onPageScroll handler with reanimated

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Summary

This PR adds a document section to integrate reanimated event handler with onPageScroll. Ideally it would be more convenient to export a hook `useAnimatedPageScrollHandler` but it'll add dependency on reanimated. I think creating a separate package for this hook can also be convenient!

## Test Plan
- I have [created a gist](https://gist.github.com/intergalacticspacehighway/52252474872c9e2246e398179920ccf2) which can be tested in RN example app which has reanimated installed and configured. The callback receives offset and position events. 
- Also this [repo](https://github.com/intergalacticspacehighway/pager-view-reanimated-repro) (App.js) includes the working example which can be referred.
- I am not really sure if there are any more cases.

### What's required for testing (prerequisites)?
- react-native-pager-view and react-native-reanimated

### What are the steps to reproduce (after prerequisites)?

- Run the [app](https://github.com/intergalacticspacehighway/pager-view-reanimated-repro) and try swiping pages. It should log the event or try updating values created using `useSharedValue`.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

Thanks for the great library 🙏

### CODE REVIEW

**Review by troZee — CHANGES_REQUESTED** [review]
Hey @intergalacticspacehighway 
Thank you for providing a doc improvements to the Pager. Could you add this snippet to the example folder ? It would be nice to have it inside the library example. 

Once you done it, I will merge it

**Comment by intergalacticspacehighway** [comment]
@troZee Done. Tested the example on android and iOS.


https://user-images.githubusercontent.com/23293248/148024873-9e61f9eb-ad1d-4c53-96c9-b4c50c1cc711.mov

File: example/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "react-native": "0.63.4",
     "react-native-animated-pagination-dots": "^0.1.51",
     "react-native-gesture-handler": "^1.9.0",
-    "react-native-reanimated": "^1.13.2",
+    "react-native-reanimated": "^2.3.1",
```
Comment (intergalacticspacehighway) [inline example/package.json:26]: I don't think reanimated was being used in the example so this should not cause any issues!

File: example/android/app/src/main/java/com/example/reactnativepagerview/MainApplication.java [hunk 2]
```diff
@@ -11,6 +11,8 @@
 import java.lang.reflect.InvocationTargetException;
 import java.util.List;
 import com.reactnativepagerview.PagerViewPackage;
+import com.facebook.react.bridge.JSIModulePackage; // <- add
+import com.swmansion.reanimated.ReanimatedJSIModulePackage; // <- add
```
Comment (intergalacticspacehighway) [inline example/android/app/src/main/java/com/example/reactnativepagerview/MainApplication.java:15]: These changes are from reanimated [installation steps](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/)

**Pull request merged**